opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/RuntimeEnvironment.java

```diff
@@ -1593,11 +1595,16 @@ public void setStatisticsFilePath(String path) {
      *
      * @return the framework
      */
-    public synchronized AuthorizationFramework getAuthorizationFramework() {
-        if (authFramework == null) {
-            authFramework = new AuthorizationFramework(getPluginDirectory(), getPluginStack());
+    public AuthorizationFramework getAuthorizationFramework() {
+        try {
+            authFrameworkLock.readLock().lock();
+            if (authFramework == null) {
+                authFramework = new AuthorizationFramework(getPluginDirectory(), getPluginStack());
```

Comment:
Theoretically, this can be accessed by multiple threads. Is it okay? If we truly want a singleton, then maybe we should revise this.

Reply:
true, we do not want multiple instances of `AuthorizationFramework` hanging around (for efficiency).